src/pytorch_lightning/loggers/mlflow.py

```diff
@@ -127,6 +139,7 @@ def __init__(
         tracking_uri: Optional[str] = os.getenv("MLFLOW_TRACKING_URI"),
         tags: Optional[Dict[str, Any]] = None,
         save_dir: Optional[str] = "./mlruns",
+        log_model: Optional[bool] = False,
```

Comment:
The type here needs to reflect the different types of choices listed in the docs above. So I suggest to go for 

```suggestion
        log_model: Literal[True, False, "all"] = False,
```